taf/git.py

```diff
@@ -180,6 +188,8 @@ def log_prefix(self) -> str:
     def pygit_repo(self) -> Optional[pygit2.Repository]:
         try:
             return self.pygit.repo
+        except PygitError as e:
```

Comment:
`self.pygit` will raise an error if `pygit` cannot be instantiated for whatever reason. I don't think that we want to catch it here. Move the logging to `self.pygit` (so log and then raise an error). I'd actually remove this `try-except` - we want the errors raised by `self.pygit` to be propagated.
There should be a test that I sent you over slack, copy-paste it to `test_git` and make sure it's passing